src/ore/src/metrics.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License in the LICENSE file at the
+// root of this repository, or online at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Metrics for materialize systems.
+//!
+//! The idea here is that each subsystem keeps its metrics in a scoped-to-it struct, which gets
+//! registered (once) to the server's (or a test's) prometheus registry.
+//!
+//! Instead of using prometheus's (very verbose) metrics definitions, we rely on type inference to
+//! reduce the verbosity a little bit. A typical subsystem will look like the following:
+//!
+//! ```rust
+//! # use ore::metrics::{MetricsRegistry, UIntCounter};
+//! # use ore::metric;
+//! #[derive(Debug, Clone)] // Note that prometheus metrics can safely be cloned
+//! struct Metrics {
+//!     pub bytes_sent: UIntCounter,
+//! }
+//!
+//! impl Metrics {
+//!     pub fn register_into(registry: &MetricsRegistry) -> Metrics {
+//!         Metrics {
+//!             bytes_sent: registry.register(metric!(
+//!                 name: "mz_pg_sent_bytes",
+//!                 help: "total number of bytes sent here",
+//!             )),
+//!         }
+//!     }
+//! }
+//! ```
+
+use prometheus::core::{
+    Atomic, Collector, GenericCounter, GenericCounterVec, GenericGauge, GenericGaugeVec, Opts,
+};
+use prometheus::proto::MetricFamily;
+use prometheus::{HistogramOpts, Registry};
+
+pub use prometheus::Opts as PrometheusOpts;
+pub use prometheus::{
+    Counter, CounterVec, Gauge, Histogram, HistogramVec, IntCounter, IntCounterVec, IntGauge,
+    IntGaugeVec, UIntCounter, UIntCounterVec, UIntGauge, UIntGaugeVec,
+};
+
+mod delete_on_drop;
+pub use delete_on_drop::*;
+
+/// Buckets that can capture data between one microsecond and 1 second.
+pub const HISTOGRAM_BUCKETS: [f64; 17] = [
+    0.000_016, 0.000_032, 0.000_064, 0.000_128, 0.000_256, 0.000_512, 0.001, 0.002, 0.004, 0.008,
+    0.016, 0.032, 0.064, 0.128, 0.256, 0.512, 1.0,
+];
```

Comment:
this reverts the recent change to shift the buckets to be a bit coarser.

Reply:
! Eep, I gotta port that. Good catch.

Reply:
(Set it up to use `ore::stats::HISTOGRAM_BUCKETS` now)